combat-trainer.lic

```diff
@@ -2298,9 +2313,156 @@ class TrainerProcess
+
+  def night_center
+    fput('center telescope on heart')
+    while line = get
+      if line.include?("That's a bit tough to do when you can't see the sky")
+        stow_telescope
+        pause 1
+        return
+      end
+      if line.include?('Your search for the constellation of the Heart')
+        day_center
+      end
+      if line.include?('You are a bit too busy performing to do that.')
+        fput('stop play')
+        night_center
+      end
+      observe
+      break
+    end
+  end
+
+  def observe
+    echo 'Observing'
+    case DRC.bput('peer telescope', 'The pain is too much', 'You learned something useful from your observation', "While the sighting wasn't quite", 'Clouds obscure the sky where the Sun should appear', 'Clouds obscure the sky where the Heart should appear', 'You peer aimlessly through your telescope', 'You have not pondered your last observation sufficiently', 'You see nothing regarding the future', "You believe you've learned all that you can about survival", 'Too many futures cloud your mind - you learn nothing.')
+    when 'You learned something useful from your observation'
+      waitrt?
+      stow_telescope
+      return
+    when 'The pain is too much'
+      stow_telescope
+      return
+    when "While the sighting wasn't quite"
+      waitrt?
+      stow_telescope
+      return
+    when 'Clouds obscure the sky where the Sun should appear'
+      stow telescope
+      return
+    when 'Clouds obscure the sky where the Heart should appear'
+      stow_telescope
+      return
+    when 'You peer aimlessly through your telescope'
+      determine_time
+    when 'You have not pondered your last observation sufficiently'
+      waitrt?
+      stow_telescope
+      return
+    when 'You see nothing regarding the future'
+      waitrt?
+      pause 1
+      observe
+    when "You believe you've learned all that you can about survival"
+      waitrt?
+      stow_telescope
+      check_predict
+    when 'Too many futures cloud your mind - you learn nothing.'
+      waitrt?
+      stow_telescope
+      check_predict
+    end
+  end
+
+  def stow_telescope
+    case DRC.bput('stow my telescope', 'Stow what', 'You put your telescope')
+    when 'You put your telescope'
+      pause 1
+      return
+    when 'Stow what'
+      return
+    end
+  end
+
+  def check_predict
     waitrt?
-    fput('engage')
+    pause 1
+    if DRSkill.getxp('Astrology') > 25
+      return
+    else
+      fput("align #{@astrology_prediction_skills_survival}")
+      waitrt?
+      pause 1
+      execute_predict
+    end
+  end
+
+  def execute_predict
+    retreat
+    case DRC.bput('predict future', 'You are far too occupied by present matters to immerse yourself in matters of the future', 'You look inside yourself')
+    when 'You are far too occupied by present matters to immerse yourself in matters of the future'
+      retreat
+      waitrt?
+      execute_predict
+    when 'You look inside yourself'
+      stow_hands
```

Comment:
❌ DRCI needs to be included to find `stow_hands` otherwise its an error

Reply:
Fixed that